testing/README.md

```diff
@@ -0,0 +1,390 @@
+
+  ```js
+  suite('MyComponent', () => {
+    suite('.staticMethod', () => {
+      test('creates a new instance', () => {
+        assert.instanceOf(MyComponent.staticMethod(), MyComponent);
+      });
+    });
+
+    suite('#instanceMethod', () => {
+      test('returns the expected value', () => {
+        assert.equal(component.instanceMethod(), 'yes');
+      });
+    });
+  });
+  ```
+
+- In general, each test has three steps: setup, perform the action under test, and assert based on the result.
+
+- If your tests involve stubbing a lot of different objects, it can be a sign of a fragile technical design. Instead of having many external dependencies that are directly brought in by the module, it can be useful to rely on [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection) instead. This design pattern can make code easier to understand and test.
+
+- Avoid using mocks.
+
+  > Why? They reverse the typical order of tests (setup, do something, assertion), which makes them harder to read in the context of other tests. Generally, you can achieve the same thing with stubs instead.
+
+  ```js
+  // bad
+  const mock = sinon.mock(subject).expects('myMethod').once();
+  subject.doSomething();
+  mock.verify();
+
+  // good
+  const stub = sinon.spy(subject, 'myMethod');
+  subject.doSoemthing();
+  assert.calledOnce(stub);
+  ```
+
+[↑ scrollTo('#table-of-contents')](#table-of-contents)
+
+
+
+## Types of testing
+
+The guide above is mostly focused on unit tests, where a "unit" of code (ideally, the smallest version of an API) is tested in isolation from the rest of the system. However, there are a few other types of testing that are important to consider when writing JavaScript: integration testing, functional testing, and UI testing.
+
+### Integration testing
+
+Integration tests ensure that various parts of a system work properly together. It mostly focuses on the design of an overall architecture rather than individual pieces. In writing integration tests, make sure to keep the following in mind:
+
+- Be careful not to do unit tests for the pieces of the system and integration tests for the system as a whole if the pieces are not part of the public interface. That is, if the pieces will only be used in combination as part of this system, the system is the actual unit of tests, and the pieces are an implementation detail that does not need direct tests (they are, of course, indirectly tested by the testing of the public interface to the system).
+
+### Functional testing
```

Comment:
@m-ux do you maybe want to add some more notes about browser/ UI tests here? I don't have a ton of experience with them.